play-json/shared/src/main/scala/Json.scala

```diff
@@ -113,23 +109,71 @@ object Json {
    * }
    * }}}
    *
-   * @param json the JsValue to convert
-   * @return a String with the json representation
+   * $jsonParam
+   * $returnStringRepr.
    */
-  def prettyPrint(json: JsValue): String = StaticBinding.prettyPrint(json)
+  def prettyPrint(json: JsValue): String
 
   /**
-   * Provided a Writes implicit for its type is available, convert any object into a JsValue.
+   * Converts any writeable value to a [[JsValue]].
+   *
+   * A value is writeable if a [[Writes]] implicit is available for its type.
    *
-   * @param o Value to convert in Json.
+   * @param o the value to convert as JSON
    */
-  def toJson[T](o: T)(implicit tjs: Writes[T]): JsValue = tjs.writes(o)
+  def toJson[T](o: T)(implicit tjs: Writes[T]): JsValue
+
+  /**
+   * Converts a [[JsValue]] to a value of requested type `T`.
+   *
+   * @tparam T The type of conversion result,
+   * only supported if a [[Reads]] implicit is available for.
+   *
+   * $jsonParam
+   */
+  def fromJson[T](json: JsValue)(implicit fjs: Reads[T]): JsResult[T]
 
   /**
-   * Provided a Reads implicit for that type is available, convert a JsValue to any type.
+   * Returns a [[JsObject]] with given fields.
    *
-   * @param json Json value to transform as an instance of T.
+   * @param fields the object fields specified as pairs of name and value
    */
+  def obj(fields: (String, Json.JsValueWrapper)*): JsObject
+
+  /** Returns a [[JsArray]] with given items. */
+  def arr(items: Json.JsValueWrapper*): JsArray
+
+}
+
+/**
+ * Helper functions to handle JsValues.
+ *
+ * @define macroOptions @tparam O the compile-time options
+ * @define macroWarning If any missing implicit is discovered, compiler will break with corresponding error.
+ * @define formatDescription Creates a `OFormat[T]` by resolving, at compile-time, the case class fields or sealed family, and the required implicits.
+ */
+object Json extends JsonFacade {
+
+  def parse(input: String): JsValue = StaticBinding.parseJsValue(input)
```

Comment:
Sure. I just thought it was weird to call `Json` object methods there, but looks like you fixed it. Probably makes more sense to keep `JsonFacade` methods abstract.